src/tiled/flipmapobjects.cpp

```diff
@@ -39,14 +40,39 @@ FlipMapObjects::FlipMapObjects(MapDocument *mapDocument,
     setText(QCoreApplication::translate("Undo Commands",
                                         "Flip %n Object(s)",
                                         nullptr, mapObjects.size()));
+
+    //computing objects center
+    QRectF boundaryObjectsRect;
+    for (MapObject *object : mMapObjects) {
+        QTransform objectTransform;
+        objectTransform.translate(object->x(), object->y());
+        objectTransform.rotate(object->rotation());
+        objectTransform.translate(-object->x(), -object->y());
+
+        if (!object->cell().isEmpty()) { //computing bound rect for cell
+            QRectF cellRect = QRectF(object->x(),
+                                     object->y(),
+                                     object->width(), -object->height()).normalized();
+            boundaryObjectsRect = boundaryObjectsRect.united(objectTransform.mapRect(cellRect));
+        } else if (!object->polygon().empty()) { //computing bound rect for polygon
+            const QPolygonF &objectPolygon = object->polygon();
+            QTransform polygonToMapTransform;
+            polygonToMapTransform.translate(object->x(),
+                                            object->y());
+            polygonToMapTransform.rotate(object->rotation());
+            boundaryObjectsRect = boundaryObjectsRect.united(polygonToMapTransform.mapRect(QRectF(objectPolygon.boundingRect())));
+        } else { //computing bound rect for other
+            boundaryObjectsRect = boundaryObjectsRect.united(objectTransform.mapRect(object->bounds()));
+        }
+    }
+    mObjectsCenter = boundaryObjectsRect.center();
 }
 
 void FlipMapObjects::flip()
 {
-    // TODO: Flip them properly as a group
-    const auto &objects = mMapObjects;
```

Comment:
Please keep this line since using the `const` alias can prevent a deep copy of the (implicitly shared) Qt container.